### PR TITLE
(SIMP-2342) Refactor svckill to use moduledata

### PR DESCRIPTION
**Pull request by heliocentric**
* Refactor svckill to use moduledata, and use
  hiera to pull the default ignore list. This
  allows us to use any fact with minimal effort
  in the service list.

### CODE REVIEW

**Comment by heliocentric** [comment]
removed modelines as that's not a fight I care about having right now.

File: manifests/init.pp [hunk 1]
```diff
@@ -56,17 +56,22 @@
...
+  Array[String]               $ignore_defaults = [],
   Array[Stdlib::Absolutepath] $ignore_files = [],
-  Boolean                     $verbose      = true
+  Boolean                     $verbose      = true,
+  Boolean                     $debug        = false
 ){
   include '::svckill::ignore::collector'
-
+  $combined_ignore_list = $ignore + $ignore_defaults
+  if ($debug == true) {
```
Comment (op-ct) [inline manifests/init.pp]: I can see why you handled the debugging this way, but it's a bit weird: 
*  There are now two class parameters for different types of verbosity 
* `$debug` logs its output using a different mechanism than `$verbose`.
* `$debug`'s method of logging adds resources to the catalog 

**A suggestion:** Handle this reporting in the `svckill` type's `change_to_s` method (the same place `$verbose` is handled) and log it using `Puppet.debug` (which reports output only when you run in `--debug` mode).  

* Both lists would be logged using the same mechanism
* You won't need to add the extra `$debug` parameter just to add a new tier of verbosity
* Logging won't affect the catalog.

File: manifests/init.pp [hunk 2]
```diff
@@ -56,17 +56,22 @@
 #
 class svckill (
   Array[String]               $ignore       = [],
+  Array[String]               $ignore_defaults = [],
   Array[Stdlib::Absolutepath] $ignore_files = [],
-  Boolean                     $verbose      = true
+  Boolean                     $verbose      = true,
```
Comment (op-ct) [inline manifests/init.pp]: Does anyone know if there's a reason we don't also expose the `svckill` type's `mode` attribute in a class parameter like we do with `$verbose`?

File: manifests/init.pp [hunk 3]
```diff
@@ -56,17 +56,22 @@
...
+  Boolean                     $debug        = false
```
Comment (op-ct) [inline manifests/init.pp]: No strings documentation for the new `$debug` parameter.

File: data/family_release/RedHat-7.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+svckill::ignore_defaults:
+  - autovt
+  - goferd
+  - ^systemd-.*
```
Comment (heliocentric) [inline data/family_release/RedHat-7.yaml]: the systemd one yes, there are 3 or 4 systemd services, and service templates in play on rhel7

File: data/common.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  - auditd
+  - puppet
+  - puppetmaster
+  - crond
+  - sshd
+  - iptables
+  - ip6tables
+  - ebtables
+  - rc
+  - ^pe-.*
```
Comment (heliocentric) [inline data/common.yaml]: the PE one I was explicitly told to use. The redhat-7 ones were to hit service template service names if they existed (which this type is not exactly handling all that well to start with, but then again neither does puppet... yay systemd!)

File: data/virtual/kvm.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+svckill::ignore_defaults:
+  - ^ovirt-.*
```
Comment (heliocentric) [inline data/virtual/kvm.yaml]: yes, ovirt-guest-agent and qemu-guest-agent are on my ovirt vms used for testing, and yes that's the point. 
the qemu-guest-agent is only on libvirt vms, while ovirt-guest-agent is also on ovirt vms; but there is no fact to differentiate the two.

Comment (heliocentric) [inline data/virtual/kvm.yaml]: I'll make those two explicit

File: manifests/init.pp (hunk 1 above)
Comment (heliocentric) [inline manifests/init.pp]: I didn't want to make a lot of changes to that ruby code, and the point of this debugging line is to debug the hiera based ignore list functionality, not the type/provider. ie, to see if the consolidation of the module data into a single array is including /all/ of the intended ignored resources.

I have a number of issues with how this type/provider work and how they send output, which I didn't want to get into. I mainly left it in precisely to ease debugging.

File: lib/puppet/provider/svckill/kill.rb [hunk 7]
```diff
@@ -141,7 +96,7 @@ def mode
       # This command is what actually checks the service on the
       # Skip anything that we're supposed to ignore
       if ignore.index{|x| Regexp.new("^#{x}$").match(obj_name) } then
-        Puppet.debug("svckill: Ignoring '#{obj_name}' due to ignore list")
+        Puppet.debug("svckill: Ignoring '#{obj_name}' due to ignore list YYY")
```
Comment (trevor-vaughan) [inline lib/puppet/provider/svckill/kill.rb]: Am I missing something special about this YYY?

File: hiera.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+---
+version: 4
+datadir: data
+hierarchy:
+  - name: "OSFamily + Release"
+    backend: "yaml"
+    path: "family_release/%{facts.osfamily}-%{facts.operatingsystemmajrelease}"
```
Comment (trevor-vaughan) [inline hiera.yaml]: We should switch these over to use the 'os' hash as it's the #newhotness

**Comment by trevor-vaughan** [comment]
I have one more patch to upload which I'll push in this evening. It fixes the report processor `to_yaml` issue.

**Pull request merged**